### ISSUE TITLE
Support Container Tabs: Local Storage

### ISSUE THREAD

**Issue by Lusito** [Lusito/forget-me-not#13]
https://wiki.mozilla.org/Security/Contextual_Identity_Project/Containers

Containers keep cookies, localstorage, etc. in different stores (named by the cookie stored) if this feature is enabled. The cookie part of this is to be solved with #8.

This ticket is about local storage, since browsingData.remove does not have an option to remove local-storage based on a cookie store id yet. There is a ticket, which would resolve this though:
https://bugzilla.mozilla.org/show_bug.cgi?id=1353726

As soon as that ticket is solved, work can be done to fix this.

**Comment by felixlang** [comment]
Is this the reason why Forget Me Not doesn't delete local storage when leaving a site, even when that rule is in place?

For example, I have a rule to cleanup *.Youtube.com On Leave after 10 seconds, with Cookies and Local Storage enabled for removal in the FMN options menu. I visit Youtube.com and then close the tab. 10 seconds later all the cookies from Youtube.com are deleted, but there is still 118kb of Local Storage left over.

**Comment by Lusito** [comment]
No, this is purely about the container tabs feature. Local Storage should be cleaned per-domain on leave.

Your issue is, that Mozilla has not been clearly naming things. For example the "Manage cookies and site data" lists "Storage", which might just be "Indexed DB", which can't be cleaned per-domain yet (this is a limitation of Firefox). You can confirm this by running manual cleanup and selecting only "indexed db". After that, the 118kb should be gone.

**Comment by felixlang** [comment]
Thank you for your quick reply and for the assistance.

For the *.youtube.com domain at least, the 118 KB of leftover data appears to be a mix of Index DB, Service Workers, and some other type of data that FMN is unable to clean.

I used manual cleanup with every option checked (and all Youtube tabs closed), which leaves 64.0 KB of the original 118 KB remaining. The only apparent way to delete the remaining 64.0 KB is to manually delete it in Firefox's options menu.
